test/dal/connection.spec.ts

```diff
@@ -0,0 +1,67 @@
+import {getConnectionDetails} from '../../src/dal/connection';
+import {expect} from 'chai';
+
+describe('DB connection logic', () => {
+  describe('getConnectionDetails()', () => {
+    const env = Object.assign({}, process.env);
+
+    after(() => {
+      process.env = env;
+    });
+
+    beforeEach(() => {
+      process.env = {
+        'DB_HOSTNAME': 'DB_HOSTNAME',
+        'DB_PORT': 'DB_PORT',
+        'DB_USERNAME': 'DB_USERNAME',
+        'DB_PASSWORD': 'DB_PASSWORD',
+        'DB_NAME': 'DB_NAME',
+      };
+    });
+
+    it('should get count of submissions', async () => {
+      const details = getConnectionDetails();
+      expect(details.host).to.equal('DB_HOSTNAME');
+      expect(details.port).to.equal('DB_PORT');
+      expect(details.user).to.equal('DB_USERNAME');
+      expect(details.password).to.equal('DB_PASSWORD');
+      expect(details.database).to.equal('DB_NAME');
+      expect(details.connectionLimit).to.equal(10);
+    });
+
+    it('should return error if missing host', async () => {
+      delete process.env.DB_HOSTNAME;
+      expect(() => {
+        getConnectionDetails();
+      }).to.throw('database connection details missing');
```

Comment:
We could create constant that can be used in the business logic when DB connection fail.